x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -538,6 +538,7 @@ export function parseDataStreamElasticsearchEntry(
     parsedElasticsearchEntry.privileges = expandedElasticsearch.privileges;
   }
 
+  // we parse the settings from the data stream, what abou tthe values from the index template?
```

Comment:
Sorry I'm not following this one. Could you provide any more info?

Reply:
To decide what to show to the user, we parse the value of the current data stream setting. But what if through a package upgrade the value of the template has been adjusted but the data stream still has the old value. This partially goes back to the above discussion around triggering a rollover when this happens which would resolve this mostly.

The second missing part also what we touched on above: There is no way currently to tell which template applies to a data stream. We know it today because of the convention but if we would have templates per namespace too, it gets trickier. An ES API for it would help here.

The more I think about all these scenarios the more I think all this handling should only be temporary in Fleet itself but be done in Elasticsearch itself. Tell Elasticsearch to change a setting for a data stream and it will do all the updates and rollovers needed ...